src/dialog-context.tsx

```diff
@@ -0,0 +1,112 @@
+'use client';
+import * as React from 'react';
+
+import type {AlertProps, ConfirmProps, DialogProps, ExtendedDialogProps} from './dialog';
+
+const DialogContext = React.createContext<{
+    mounted: boolean;
+    dialog: DialogProps | null;
+    setDialog: React.Dispatch<React.SetStateAction<DialogProps | null>>;
+}>({mounted: false, dialog: null, setDialog: () => {}});
+
+const ModalDialog = React.lazy(() => import(/* webpackChunkName: "dialog" */ './dialog'));
+
+export const useDialog = (): {
+    alert: (params: AlertProps) => void;
+    confirm: (params: ConfirmProps) => void;
+    dialog: (params: ExtendedDialogProps) => void;
+} => {
+    const {dialog: currentDialog, setDialog} = React.useContext(DialogContext);
+
+    const showDialog = React.useCallback(
+        (params: DialogProps) => {
+            if (params && currentDialog) {
+                throw Error('Tried to show a dialog on top of another dialog');
+            }
+            setDialog(params);
+        },
+        [setDialog, currentDialog]
+    );
+
+    return React.useMemo(
+        () => ({
+            alert: (params: AlertProps) => showDialog({type: 'alert', ...params}),
+            confirm: (params: ConfirmProps) => showDialog({type: 'confirm', ...params}),
+            dialog: (params: ExtendedDialogProps) => showDialog({type: 'dialog', ...params}),
+        }),
+        [showDialog]
+    );
+};
+
+const throwMissingDialogRootError = () => {
+    throw Error(
+        'Tried to show a dialog but the DialogRoot component was not mounted (missing ThemeContextProvider?)'
+    );
+};
+
+const missingDialogFunctions: ReturnType<typeof useDialog> = {
+    alert: throwMissingDialogRootError,
+    confirm: throwMissingDialogRootError,
+    dialog: throwMissingDialogRootError,
+};
+
+let currentDialogFunctions = missingDialogFunctions;
+
+/**
+ * @deprecated Created for backwards compatibility
+ */
+const ExposeDialogFunctions = (): JSX.Element => {
+    const dialogFunctions = useDialog();
+    React.useEffect(() => {
+        currentDialogFunctions = dialogFunctions;
+        return () => {
+            currentDialogFunctions = missingDialogFunctions;
+        };
+    }, [dialogFunctions]);
+    return <></>;
+};
+
+/**
+ * @deprecated Use useDialog to get this function
+ */
+export const alert = (params: AlertProps): void => currentDialogFunctions.alert(params);
+
+/**
+ * @deprecated Use useDialog to get this function
+ */
+export const confirm = (params: ConfirmProps): void => currentDialogFunctions.confirm(params);
+
+/**
+ * @deprecated Use useDialog to get this function
+ */
+export const dialog = (params: ExtendedDialogProps): void => currentDialogFunctions.dialog(params);
```

Comment:
I've deprecated this because you should get those function from the `useDialog` hook, but I find this very convenient so I'm not sure whether to keep or remove the deprecation :/